receiver/datadogreceiver/httpmetrics.go

```diff
@@ -0,0 +1,46 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package datadogreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/datadogreceiver"
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"go.opentelemetry.io/otel/metric"
+	semconv "go.opentelemetry.io/otel/semconv/v1.25.0"
+)
+
+func httpMetrics(meter metric.Meter, next http.Handler) http.Handler {
```

Comment:
I think we might have a similar metric already being created for all HTTP servers already, perhaps shown only with the telemetry level is set to detailed. If not, we might want to use mdatagen to define the metric.